dask_sql/physical/rex/core/call.py

```diff
@@ -634,7 +635,10 @@ def to_timestamp(self, df, format):
         else:
             if format != default_format:
                 raise RuntimeError("Integer input does not accept a format argument")
-            return dd.to_datetime(df, unit="s")
+            if is_cudf_type(df):
+                return df
```

Comment:
Looking into the GPU failures, it seems like in some cases by skipping over the `to_datetime` call here means we end up passing an integer series to `convert_to_datetime` and erroring.

Is there a particular case where we wouldn't want to call `to_datetime` here?

Reply:
Yes, the case that fails is
```
df = pd.DataFrame({"d1": [1203073300], "d2": [1503073700]})
c.create_table("df", df, gpu=gpu)
expected_df = pd.DataFrame({"dt": [3472]})
df1 = c.sql(
    "SELECT TIMESTAMPDIFF(DAY, to_timestamp(d1), to_timestamp(d2)) AS dt FROM df"
)
df1.compute()
```
which returns 0 instead of 3472 in the GPU case when `to_datetime` is used here. I'm not really sure why this happens, and there wasn't an obvious type check to avoid this. But you're right, it makes more sense to call `to_datetime` either way, so for now we can just skip that case.